data_model/src/events/data/events.rs

```diff
@@ -0,0 +1,301 @@
+//! This module contains data events
+
+use super::*;
+
+/// Trait for retrieving id from events
+pub trait IdTrait: Identifiable {
+    fn id(&self) -> &Self::Id;
+}
```

Comment:
Retrieving is okay,
but generally I don't think an event is `Identifiable` by the id of the entity that the event describes